universal-application-tool-0.0.1/test/controllers/HomeControllerAuthenticatedTest.java

```diff
@@ -30,8 +39,24 @@ protected Application provideApplication() {
 
   @Before
   public void setUp() {
+    // Get the config, and hack it so that all requests appear authorized.
     Config config = app.injector().instanceOf(Config.class);
-    config.setClients(new Clients(AnonymousClient.INSTANCE));
+    AnonymousClient client = AnonymousClient.INSTANCE;
+    config.setClients(new Clients(client));
+
+    // The SecurityLogic wants to use some smarts to figure out which client to use, but
+    // those smarts are never going to pick this client (since none of the endpoints are
+    // configured to use it), so we implement an anonymous client finder which always returns
+    // our client.
+    DefaultSecurityLogic securityLogic = new DefaultSecurityLogic();
```

Comment:
question: Is this something we will need to do for every controller test? If so, should we (in a future PR) refactor this out so all the controller tests can inherit the same logic?

Reply:
Definitely, that was my intent for this class in the medium term.  For now I figure we'll just put all those tests in here.